testnet/stacks-node/src/tests/signer/v0.rs

```diff
@@ -5170,3 +5170,285 @@ fn signing_in_0th_tenure_of_reward_cycle() {
     }
     assert_eq!(signer_test.get_current_reward_cycle(), next_reward_cycle);
 }
+
+/// This test involves two miners with a custom chain id, each mining tenures with 6 blocks each.
+/// Half of the signers are attached to each miner, so the test also verifies that
+/// the signers' messages successfully make their way to the active miner.
+#[test]
+#[ignore]
+fn multiple_miners_with_custom_chain_id() {
+    let num_signers = 5;
+    let max_nakamoto_tenures = 20;
+    let inter_blocks_per_tenure = 5;
+
+    // setup sender + recipient for a test stx transfer
+    let sender_sk = Secp256k1PrivateKey::new();
+    let sender_addr = tests::to_addr(&sender_sk);
+    let send_amt = 1000;
+    let send_fee = 180;
+    let recipient = PrincipalData::from(StacksAddress::burn_address(false));
+
+    let btc_miner_1_seed = vec![1, 1, 1, 1];
+    let btc_miner_2_seed = vec![2, 2, 2, 2];
+    let btc_miner_1_pk = Keychain::default(btc_miner_1_seed.clone()).get_pub_key();
+    let btc_miner_2_pk = Keychain::default(btc_miner_2_seed.clone()).get_pub_key();
+
+    let node_1_rpc = 51024;
+    let node_1_p2p = 51023;
+    let node_2_rpc = 51026;
+    let node_2_p2p = 51025;
```

Comment:
Didn't one of our esteemed engineers write a `gen_random_port` function recently? Can't remember...

At first I wasn't sure if this was needed to preserve the even/odd thing, but since that just depends on the signer ports, these can be random I think?

Reply:
yes they can XD this is a copy paste from the other test which is not randomized...I will fix XD